packages/wallets/banana/src/adapter.ts

```diff
@@ -0,0 +1,135 @@
+
+    private _connecting: boolean;
+    private _wallet: BananaWallet | null;
+    private _publicKey: PublicKey | null;
+    private _readyState: WalletReadyState =
+        typeof window === 'undefined' || typeof document === 'undefined'
+            ? WalletReadyState.Unsupported
+            : WalletReadyState.NotDetected;
+
+    constructor(config: BananaWalletAdapterConfig = {}) {
+        super();
+        this._connecting = false;
+        this._wallet = null;
+        this._publicKey = null;
+
+        if (this._readyState !== WalletReadyState.Unsupported) {
+            const checkIsBananaAvailable = async () => {
+                const isBananaWalletAvailable = await checkIsAvailable(config);
+                if (isBananaWalletAvailable) {
+                    this._readyState = WalletReadyState.Installed;
+                    this.emit('readyStateChange', this._readyState);
+                    return true;
+                }
+                return false;
+            };
+            checkIsBananaAvailable();
+        }
```

Comment:
Sorry, loading external libraries must only be done asynchronously inside the `connect` function. Otherwise you're making every app user load this library regardless of whether they use the wallet. Please check out the other `Loadable` adapters for how this is done.